test/integration/integration_test.go

```diff
@@ -0,0 +1,310 @@
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+	} {
+		if agentMock.Has(&event) {
+			continue
+		}
+
+		e := json.NewEncoder(os.Stderr)
+		t.Log("Expected:")
+		_ = e.Encode(event)
+		t.Log("Have:")
+		_ = e.Encode(agentMock.Events())
+		t.Fatalf("Event was not captured")
+	}
+}
+
+func Test_Sink_receives_common_Pod_deletion_events(t *testing.T) {
+	client, agentMock := initialize(t)
+
+	t.Log("Creating test namespace...")
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: nsName(t),
+		},
+	}
+	ns, err := client.CoreV1().Namespaces().Create(contextFromTestDeadline(t), ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("could not create %s namespace: %v", ns, err)
+	}
+
+	t.Cleanup(func() {
+		t.Log("Cleaning up test namespace...")
+		err := client.CoreV1().Namespaces().Delete(contextFromTestDeadline(t), ns.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Logf("could not delete test namespace '%s': %v", ns.Name, err)
+		}
+	})
+
+	t.Log("Creating test pod...")
+	testpod, err := client.CoreV1().Pods(ns.Name).Create(contextFromTestDeadline(t), &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nginx-e2e-killable",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "nginx",
+					Image: "nginx",
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("could not create test pod: %v", err)
+	}
+
+	time.Sleep(7 * time.Second)
+
+	err = client.CoreV1().Pods(ns.Name).Delete(contextFromTestDeadline(t), testpod.Name, metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("could not create test pod: %v", err)
+	}
+
+	t.Log("Waiting for events to show up...")
+	agentMock.Wait(15*time.Second, 1*time.Minute)
+	for _, event := range []sdkEvent.Event{
+		{
+			Summary:  "Stopping container " + testpod.Spec.Containers[0].Name,
+			Category: "kubernetes",
+			Attributes: map[string]interface{}{
+				"event.metadata.name":             testpod.Name + ".",
+				"event.metadata.namespace":        ns.Name,
+				"event.reason":                    "Killing",
+				"clusterName":                     "",
+				"event.involvedObject.apiVersion": "",
+				"event.involvedObject.kind":       "Pod",
+				"event.involvedObject.name":       testpod.Name,
+				"event.message":                   "Stopping container " + testpod.Spec.Containers[0].Name,
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+	} {
+		if agentMock.Has(&event) {
+			continue
+		}
+
+		e := json.NewEncoder(os.Stderr)
+		t.Log("Expected:")
+		e.Encode(event) // nolint:errcheck
+		t.Fatalf("Event was not captured")
+	}
+}
+
+// nsName performs basic sanitization on the test name to convert it to an acceptable namespace name.
+func nsName(t *testing.T) string {
+	t.Helper()
+
+	return "e2e-" + strings.NewReplacer("_", "-").Replace(strings.ToLower(t.Name()))
```

Comment:
Using replacer here seems like overkill :smile:

Reply:
Yeah, i thoigh I was going to need to replace two things but in the end one was enough :)
I'll change it to the regular replace